### ISSUE TITLE
incompatible network structure and pre-trained weights

### ISSUE THREAD

**Issue by serengil** [leondgarse/Keras_insightface#24]
Hello,

I load the model shared here: https://drive.google.com/file/d/1EoYQytka3w7EeTh1v9WioBNxolGnPWp6/view?usp=sharing and then I saved just the weights with save_weights. Then, build the resnet34 model as shown below and load just weights.

basic_model = train.buildin_models("r34", dropout=0.4, emb_shape=512, output_layer='E', bn_momentum=0.9, bn_epsilon=2e-5)
basic_model.load_weights("weights.h5")

This action causes a trouble in the embedding layer: the new layer expects 3 weights, but the saved weights have 4 elements.

I found the solution. In [train.py](https://github.com/leondgarse/Keras_insightface/blob/master/train.py) 124th line

`embedding = keras.layers.BatchNormalization(momentum=bn_momentum, epsilon=bn_epsilon, name="embedding", scale=False)(nn)`

If I set scale argument to True, then this problem is solved.

The question is that did you set scale argument to True for the model you shared?

**Comment by leondgarse** [comment]
The h5 model shared in google drive is an early one, and is using `scale=True`. Then I found this difference with MXNet version after that, and trained `r34` with `scale=False`. Those plotted data is this `scale=False` one.
But this new model is deleted... Sorry I forgot this... I'll train it again later.

**Comment by serengil** [comment]
That's not a problem. I spend a while to understand that problem. I opened an issue here because some others might have this issue and they wouldn't spend time about this :)

BTW, it is mentioned that shared model has 99.40% on LFW data set. This score got when scale = True?

**Comment by leondgarse** [comment]
Ya, it may cause some trouble.
And yes, that shared model is trained using `scale=True`. It's training strategy is a little different:
- Trained 1 epoch of `softmax` to warmup.
- Using optimizer `tfa.optimizers.SGDW(learning_rate=0.1, weight_decay=5e-4, momentum=0.9)`
- Added a `l2_regularizer` to model output layer, where `l2=5e-4 * (10 - 1) / 2`. Then I found this is NOT essential, and this [adding l2_regularizer](https://github.com/leondgarse/Keras_insightface/blob/99009e89a556c5628896b987db371d6564c02ef6/train.py#L351) part is deleted.

**Comment by serengil** [comment]
So, we have pre-trained model on Google Drive and it got 99.40% on LFW. Awesome!

**Comment by leondgarse** [comment]
I trained `r34` on `CASIA` again and uploaded the model to Google drive. This time it scores higher on `agedb_30` but lower on `lfw` and `cfp_fp`. I think it's caused by random status.
